app/model/PaymentSpecificData.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model
+import play.api.libs.json._
+
+trait PaymentSpecificData {
+  def getReference: String
+}
+
+case class PaymentSpecificDataP800(
+    referencePart1: String,
```

Comment:
It would be more informative if we could name these parts according to what they are. Probably it's NINO, Year and something (chargeRef??).